src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -55,6 +55,9 @@ _log_callback (mongocrypt_log_level_t mongocrypt_log_level,
    case MONGOCRYPT_LOG_LEVEL_TRACE:
       log_level = MONGOC_LOG_LEVEL_TRACE;
       break;
+   default:
+      log_level = MONGOC_LOG_LEVEL_CRITICAL;
+      break;
```

Comment:
Also observed a warning on gcc 5.4 locally:

```
/vagrant/mongo-c-driver/src/libmongoc/src/mongoc/mongoc-crypt.c:42:4: error: switch missing default case [-Werror=switch-default]
    switch (mongocrypt_log_level) {
    ^
```

It seemed reasonable to me to consider an unknown libmongocrypt log level as critical.